### ISSUE TITLE
`Error while running: #<Errno::EIO: Input/output error @ io_fillbuf - fd:7 masterpty:/dev/pts/4>`

### ISSUE THREAD

**Issue by alexjfisher** [xeger/backticks#21]
Hi!

Since the release of `1.0.4` all of my Jenkins builds running [onceover](https://github.com/dylanratcliffe/onceover) are failing with

```
Error while running: #<Errno::EIO: Input/output error @ io_fillbuf - fd:7 masterpty:/dev/pts/4>
```

Pinning to `1.0.3` and they work again.  Is there something `onceover` is doing wrong, or is there a regression in this project?

Thanks.

**Comment by apoloval** [comment]
Same problem here! There is something buggy in that patched version 1.0.4:

```
Input/output error @ io_fillbuf - fd:7 masterpty:/dev/pts/89 (Errno::EIO)
/home/apoloval/.asdf/installs/ruby/2.6.3/lib/ruby/gems/2.6.0/gems/backticks-1.0.4/lib/backticks/command.rb:75:in `eof?'
/home/apoloval/.asdf/installs/ruby/2.6.3/lib/ruby/gems/2.6.0/gems/backticks-1.0.4/lib/backticks/command.rb:75:in `eof?'
/home/apoloval/.asdf/installs/ruby/2.6.3/lib/ruby/gems/2.6.0/gems/backticks-1.0.4/lib/backticks/command.rb:108:in `join'
```

**Comment by bartb** [comment]
+1 Same problem. Going back to 1.0.3 fixes it.

**Comment by xeger** [comment]
Argh! I'm sorry; 1.0.4 resolves a race condition with output capture and apparently, it's lacking adequate tests.

There seems to be some issue with the way `IO.select` works across operating systems; I tested on OS X and whatever Linux Travis is using, and didn't encounter `Errno::EIO` at all.

I'm flying blind here, but I will release 1.0.5 with a reasonable fix based on your issue reports.

**Comment by xeger** [comment]
I've published 1.0.5 with a fix for this issue; thankfully the [root cause is well known](https://stackoverflow.com/questions/10238298/ruby-on-linux-pty-goes-away-without-eof-raises-errnoeio) which gives me more confidence.

If you see any further errors, don't hesitate to reopen or file a new issue. Many thanks for being real-world testers!

**Comment by bartb** [comment]
Thanks for the quick fix! It works again for rhel7/ruby2.7
